File: binding/src/charts/libqlineseries.cpp
#include <QLineSeries>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QXYSeries>
#include <qlineseries.h>
#include "libqlineseries.h"
#include "libqlineseries.hxx"

QLineSeries* QLineSeries_new() {
	 return new VirtualQLineSeries();
}

QLineSeries* QLineSeries_new2(QObject* parent) {
	 return new VirtualQLineSeries(parent);
}

libqt_string QLineSeries_Tr(const char* s) {
	QString _ret = QLineSeries::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

int QLineSeries_Type(const QLineSeries* self) {
	return self->type();
}

libqt_string QLineSeries_Tr2(const char* s, const char* c) {
	QString _ret = QLineSeries::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QLineSeries_Tr3(const char* s, const char* c, int n) {
	QString _ret = QLineSeries::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
int QLineSeries_QBaseType(const QLineSeries* self) {
	auto* vqlineseries = dynamic_cast<const VirtualQLineSeries*>(self);
	if (vqlineseries && vqlineseries->isVirtualQLineSeries) {
vqlineseries->setQLineSeries_Type_IsBase(true);
	return vqlineseries->type();
}
}

// Auxiliary method to allow providing re-implementation
void QLineSeries_OnType(const QLineSeries* self, intptr_t slot) {
	auto* vqlineseries = dynamic_cast<const VirtualQLineSeries*>(self);
	if (vqlineseries && vqlineseries->isVirtualQLineSeries) {
vqlineseries->setQLineSeries_Type_Callback(reinterpret_cast<VirtualQLineSeries::QLineSeries_Type_Callback>(slot));
}
}

void QLineSeries_Delete(QLineSeries* self) {
    delete self;
}

File: binding/src/charts/libqlineseries.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qlineseries.html
pub const qlineseries = struct {

    /// New constructs a new QLineSeries object.
    pub fn New() QtC.QLineSeries {
        return qtc.QLineSeries_new();
    }


    /// New2 constructs a new QLineSeries object.
    pub fn New2(parent: ?*anyopaque) QtC.QLineSeries {
        return qtc.QLineSeries_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QLineSeries_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qlineseries.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Type(self: ?*anyopaque, ) i32 {
        return qtc.QLineSeries_Type(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnType(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) i32) void {
        qtc.QLineSeries_OnType(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseType(self: ?*anyopaque, ) i32 {
        return qtc.QLineSeries_QBaseType(@ptrCast(self));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QLineSeries_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qlineseries.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QLineSeries_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qlineseries.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QLineSeries_Delete(@ptrCast(self));
    }
};
